fem/transfer.cpp

```diff
@@ -762,6 +2126,47 @@ void L2ProjectionGridTransfer::L2ProjectionH1Space::ProlongateTranspose(
    }
 
    SetFromTDofsTranspose(fes_lor, Y, y);
+
+   if (verify_solution)
+   {
+      Vector y_temp(y.Size());
+      DeviceProlongateTranspose(x, y_temp);
+      y_temp -= y;
+      real_t error = y_temp.Norml2();
+      if (error > ho_lor_tol)
+      {
+         MFEM_VERIFY(false, "Prolongate transpose difference too high = "<<error);
+      }
+   }
+}
+
+void L2ProjectionGridTransfer::L2ProjectionH1Space::DeviceProlongateTranspose(
+   const Vector &x, Vector &y) const
+{
+   Vector X(fes_ho.GetTrueVSize());
+   Vector X_dim(pcg_vea.Width());
+   Vector Xbar(pcg_vea.Height());
+
+   Vector Y_dim(M_LH_vea->Height());
+   Vector Y(fes_lor.GetTrueVSize());
+
+   Array<int> vdofs_list;
+
+   GetTDofsTranspose(fes_ho, x, X);
+
+   for (int d = 0; d < fes_ho.GetVDim(); ++d)
+   {
+      TDofsListByVDim(fes_ho, d, vdofs_list);
+      X.GetSubVector(vdofs_list, X_dim);
+      // Compute y = P^T x = M_LH (R^T M_LH)^(-1) X = M_LH Xbar
+      Xbar = 0.0;
+      pcg_vea.Mult(X_dim, Xbar);
+      M_LH_vea->Mult(Xbar, Y_dim);
+      TDofsListByVDim(fes_lor, d, vdofs_list);
+      Y.SetSubVector(vdofs_list, Y_dim);
+   }
+
+   SetFromTDofsTranspose(fes_lor, Y, y);
 }
```

Comment:
This is basically identical to `L2ProjectionGridTransfer::L2ProjectionH1Space::ProlongateTranspose`, we don't need the duplication